### PR TITLE
Adding third part of apns

### PR DESCRIPTION
**Pull request by wenjdu**
Receiving notification deviceId and delegates it to matchMonitor to get the matches for this specific deviceId

known issue : 
- Each notification, concerns one and only one deviceId **for the moment** -> TO DO : We could have an array of deviceId. But if it is the case we need to create our own parser and build the array. I need to have a closer look at UNNotificationResponse and UNNotification.

I kept it simple, for this first try.

### CODE REVIEW

File: AlpsSDK/AlpsManager/AlpsManager.swift [hunk 1]
```diff
@@ -30,7 +30,7 @@ public class AlpsManager: MatchMonitorDelegate, ContextManagerDelegate {
     
     public lazy var contextManager = ContextManager(delegate: self)
     public lazy var matchMonitor = MatchMonitor(delegate: self)
-    public let remoteNotificationManager: RemoteNotificationManager = RemoteNotificationManager()
+    public var remoteNotificationManager: RemoteNotificationManager!
```
Comment (wenjdu) [inline AlpsSDK/AlpsManager/AlpsManager.swift]: Reversed it back to `var`, I am having issue with `self`when it is `let`

File: AlpsSDK/AlpsManager/AlpsManager.swift [hunk 2]
```diff
@@ -47,6 +47,7 @@ public class AlpsManager: MatchMonitorDelegate, ContextManagerDelegate {
         if let baseURL = baseURL {
             self.baseURL = baseURL
         }
+        remoteNotificationManager = RemoteNotificationManager(delegate: matchMonitor)
```
Comment (wenjdu) [inline AlpsSDK/AlpsManager/AlpsManager.swift:51]: At first, I wanted to use `AlpsManager` to be the delegate. 

But this is creating one more indirection so I went straight for `MatchMonitor`, what do you think ?

File: AlpsSDK/MatchMonitor.swift [hunk 3]
```diff
@@ -57,4 +58,13 @@ public class MatchMonitor {
             self.delegate?.didFind(matches: matches, for: device)
         }
     }
+    
+    // When MatchMonitor receive a notification(push or ws) it will get the matches for the monitored devices.
+    internal func remoteNotificationManager(manager: RemoteNotificationManager, didReceiveNotification: String) {
+        monitoredDevices.forEach {
+            if $0.id == didReceiveNotification {
+                getMatchesForDevice(device: $0)
+            }
```
Comment (wenjdu) [inline AlpsSDK/MatchMonitor.swift:67]: Checking the matches only for the Monitored Devices.

File: AlpsSDK/RemoteNotificationManager.swift [hunk 4]
```diff
@@ -6,18 +6,24 @@
 //  Copyright © 2017 Alps. All rights reserved.
 //
 
-import Foundation
 import UserNotifications
+import UIKit
```
Comment (wenjdu) [inline AlpsSDK/RemoteNotificationManager.swift:10]: I remember why I had to `import UIKit`

Else it doesn't reckon the `UIApplication.shared.registerForRemoteNotifications()` in registerForPushNotifications() function

So I decided to take out Foundation and keep UIKit instead

File: AlpsSDK/RemoteNotificationManager.swift [hunk 5]
```diff
@@ -6,18 +6,24 @@
...
+
+protocol RemoteNotificationManagerDelegate: class {
+    func remoteNotificationManager(manager: RemoteNotificationManager, didReceiveNotification: String)
+}
```
Comment (wenjdu) [inline AlpsSDK/RemoteNotificationManager.swift:14]: This protocol could be a general case and could work for websocket too, but I am not sure.

File: AlpsSDK/AlpsManager/AlpsManager.swift (hunk 2 above)
Comment (maciejburda) [reply AlpsSDK/AlpsManager/AlpsManager.swift:51]: Looks reasonable.

File: AlpsSDK/AlpsManager/AlpsManager.swift (hunk 1 above)
Comment (maciejburda) [reply AlpsSDK/AlpsManager/AlpsManager.swift]: I think we can change it to let without `!`. Just make sure you init it in evey AlpsManager's `init()`

File: AlpsSDK/RemoteNotificationManager.swift (hunk 4 above)
Comment (maciejburda) [reply AlpsSDK/RemoteNotificationManager.swift:10]: Make sense. Thanks for the clarification.

File: AlpsSDK/RemoteNotificationManager.swift (hunk 5 above)
Comment (maciejburda) [reply AlpsSDK/RemoteNotificationManager.swift:14]: If we don't use it - let's remove it.

**Pull request merged**